### PR TITLE
API integration users not in group

### PR DESCRIPTION
**Pull request by anoopjkottoor**

### CODE REVIEW

File: src/lib/components/dashboard/groups/AddUserToGroup.svelte [hunk 1]
```diff
@@ -3,20 +3,16 @@
     import { fetchGroupUsers, fetchAllUsers, addUserToGroup } from "../apis";
     import { User } from "../dtos";
     import { selectedGroup, showAddUserToGroupDrawer } from "../store";
+    import { fetchUsersWithoutGroupAccess } from "../../../apis/group.api";
```
Comment (AbrahamGeorge8547) [inline src/lib/components/dashboard/groups/AddUserToGroup.svelte]: please export this api in api.ts file and import from there.

File: src/lib/components/dashboard/groups/AddUserToGroup.svelte [hunk 2]
```diff
@@ -3,20 +3,16 @@
...
-        const allUsersPromise = fetchAllUsers();
-        const [userGroup, allUsers] = await Promise.all([
-            userGroupPromise,
-            allUsersPromise,
-        ]);
-        // TODO: API to fetch non group members.
-        users = allUsers.data.filter(
-            (user) =>
-                !userGroup.data.some((groupUser) => groupUser.id === user.id),
+        const userGroupPromise = fetchUsersWithoutGroupAccess(
```
Comment (AbrahamGeorge8547) [inline src/lib/components/dashboard/groups/AddUserToGroup.svelte]: use await instead of promise.all since we only have a single request.

File: src/lib/components/dashboard/groups/AddUserToGroup.svelte [hunk 3]
```diff
@@ -5,32 +5,22 @@
...
-            userGroupPromise,
-            allUsersPromise,
-        ]);
-        // TODO: API to fetch non group members.
-        users = allUsers.data.filter(
-            (user) =>
-                !userGroup.data.some((groupUser) => groupUser.id === user.id),
-        );
-
-        const credentialFieldsResponse = await fetchCredentialFieldsByGroupId(
```
Comment (AbrahamGeorge8547) [inline src/lib/components/dashboard/groups/AddUserToGroup.svelte:30]: This an important part of the code. why did you remove this?

Comment (anoopjkottoor) [reply src/lib/components/dashboard/groups/AddUserToGroup.svelte:30]: sorry missed it

**Review by AbrahamGeorge8547 — APPROVED** [review]
